### ISSUE TITLE
Contacts with HeightField

### ISSUE THREAD

**Issue by rohanpsingh** [deepmind/mujoco#236]
I think this has been discussed before in the old forum. But I haven't found a good fix as of yet.

The collisions between a box and a height field are strange (significant penetrations, sharp peaks, unexplained vectors).  
I know that the colllision detection for height fields is computed differently,
but I want to know if it is "normal" it looks like this or if there is something horribly wrong in my model.

Also, what would be way to improve the accuracy of the height field collisions without giving up too much on speed?

https://user-images.githubusercontent.com/16384313/162890700-e9f84c50-088b-4d1c-822a-c96a6859c150.mp4

Screencap showing unexpected (unrealistic?) contact force direction on the right foot:  

![frame_0000000](https://user-images.githubusercontent.com/16384313/162891767-86802002-b4d8-4008-bb40-a85019c5a104.png)

**Comment by yuvaltassa** [comment]
TL;DR make your heightfield less dense.

As mentioned [here](https://mujoco.readthedocs.io/en/latest/XMLreference.html#asset-hfield), there is a limit on the number of collisions between any two geoms, and heightfields are counted as one geom, even though they are actually a union of prisms. So contacts are getting discarded, the foot sinks in, and then contacts are found again, leading to this situation.

Basically heightfields are meant to model _macro_ terrain features rather than _micro_ features. Each "pixel" of your HF should be at least the size of a foot, ideally much bigger. If you want to simulate micro features (pebbles, things smaller than a foot) I would recommend procedurally placing sunken spheres on a plane, or something similar.
